### PR TITLE
Improve some assignment operators

### PR DESCRIPTION
**Pull request by prckent**
## Proposed changes

A few too many consts / non-enough consts found using CodeCheck/Understand.

## What type(s) of changes does this code introduce?

- Bugfix
- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?

gcc12, mac laptop only.

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'

### CODE REVIEW

**Comment by prckent** [comment]
Test this please

File: src/Utilities/ResourceHandle.h [hunk 1]
```diff
@@ -27,7 +27,7 @@ class ResourceHandle : private std::optional<std::reference_wrapper<RS>>
 
   ResourceHandle() = default;
   ResourceHandle(RS& res) { Base::emplace(res); }
-  ResourceHandle& operator=(RS& res)
+  ResourceHandle& operator=(const RS& res)
```
Comment (ye-luo) [inline src/Utilities/ResourceHandle.h:30]: This is not a conventional assignment operator. Resources needs be modifiable and thus it should not carry const. I noticed this function not being used anywhere, we can probably just delete it.

Comment (prckent) [reply src/Utilities/ResourceHandle.h]: Removed

**Comment by prckent** [comment]
Test this please

**Review by ye-luo — APPROVED** [review]

**Pull request merged**